File: src/sorting_algo/SelectionSort.java
package sorting_algo;

public class SelectionSort {
    public static void main(String[] args){

        int arr[] = { 55,64,3, 34, 25, 12, 22, 11, 90 };
        sorting(arr);

        System.out.print("Sorted array: ");
        SortedArrayPrint(arr);
    }

    static void sorting(int[] arr){
        for (int i = 0; i < arr.length-1; i++) {
            int min_index = i;
            for (int j = i+1; j < arr.length; j++) {
                if (arr[j]< arr[min_index]){
                    min_index = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[min_index];
            arr[min_index] = temp;
        }
    }

    static void SortedArrayPrint(int[] arr){
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}

File: SelectionSort.js
const SelectionSort = () => {
  let arr = [73, 57, 49, 99, 133, 20, 1, 64, 34, 25, 12, 22, 11, 90];
  let sortedArr = sorting(arr);
  console.log(sortedArr);
};

const sorting = (arr) => {
  let min_index, temp;

  for (let i = 0; i < arr.length - 1; i++) {
    min_index = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[min_index]) {
        min_index = j;
      }
    }
    temp = arr[min_index];
    arr[min_index] = arr[i];
    arr[i] = temp;
  }
  return (newArr = [...arr]);
};

SelectionSort();
